src/dbnode/storage/series/lookup/entry.go

```diff
@@ -161,6 +202,49 @@ func (entry *Entry) OnIndexFinalize(blockStartNanos xtime.UnixNano) {
 	entry.DecrementReaderWriterCount()
 }
 
+// Write writes a new value.
+func (entry *Entry) Write(
+	ctx context.Context,
+	timestamp time.Time,
+	value float64,
+	unit xtime.Unit,
+	annotation []byte,
+	wOpts series.WriteOptions,
+) (bool, series.WriteType, error) {
+	if idx := entry.indexWriter; idx != nil {
+		if entry.NeedsIndexUpdate(idx.BlockStartForWriteTime(timestamp)) {
+			entry.pendingIndexBatchSizeOne[0] = writes.PendingIndexInsert{
+				Entry: index.WriteBatchEntry{
+					Timestamp:     timestamp,
+					OnIndexSeries: entry,
+					EnqueuedAt:    entry.nowFn(),
+				},
+				Document: entry.Series.Metadata(),
+			}
+			entry.OnIndexPrepare()
+			if err := idx.WritePending(entry.pendingIndexBatchSizeOne); err != nil {
+				return false, 0, err
+			}
+		}
+	}
+	return entry.Series.Write(
+		ctx,
+		timestamp,
+		value,
+		unit,
+		annotation,
+		wOpts,
+	)
+}
+
+// LoadBlock loads a single block into the series.
+func (entry *Entry) LoadBlock(
+	block block.DatabaseBlock,
+	writeType series.WriteType,
+) error {
+	return entry.Series.LoadBlock(block, writeType)
```

Comment:
We need to do the same is indexing logic here, since the block we're loading could be not indexed.